Update frequenz-api-common to v0.5.0

tiyash-basu-frequenz opened the pull request.

tiyash-basu-frequenz committed:
Upgrade frequenz-api-common to v0.5.0

This commit upgrades the common protobuf dependency to v0.5.0.
The protobuf messages returned by the RPCs are now in the
`frequenz.api.common.v1` package.

As a result, all files besides microgrid.proto have become obsolete, and
have been removed.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Remove obsolete release notes

After upgrading to v0.5.0 of frequenz-api-common, a few older release notes
became obsolete, and therefore, have been removed.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

llucax commented on RELEASE_NOTES.md:
```diff
@@ -63,6 +63,11 @@
   `frequenz.api.microgrid.v1.<package>`. `v1` is the API's major version, and
   will be incremented for breaking changes.
 
+- The common protobuf dependency has been upgraded to `v0.5.0`. The protobuf
+  messages returned by the RPCs are now in the `frequenz.api.common.v1` package.
+  As a result, all files besides `microgrid.proto` have become obsolete, and
+  should be removed.
```
:scissors: :tada:

tiyash-basu-frequenz replied:
I actually updated the note. The files _have been_ removed.

llucax replied:
Haha, I got so excited I didn't read right through the end :rofl:

llucax commented on pyproject.toml:
```diff
@@ -26,7 +26,7 @@ classifiers = [
 requires-python = ">= 3.11, < 4"
 dependencies = [
-  "frequenz-api-common >= 0.4.0, < 0.5",
+  "frequenz-api-common@git+https://github.com/frequenz-floss/frequenz-api-common",
```
I guess this can be put back to point to a proper release  now that 0.5.0 is released.

tiyash-basu-frequenz replied:
Oops, had forgotten about this. Thanks for pointing out. Fixed.

llucax reviewed (commented):
You probably want to also update [the `mkdocs.yml` file](https://github.com/frequenz-floss/frequenz-api-microgrid/blob/e4323472fe032f4846fc914b7077291cb3a5d858/mkdocs.yml#L109) (maybe also do a `git grep api-common` to make sure we are not missing any other reference?

llucax commented:
I guess you didn't saw the review comment above?

![image](https://github.com/frequenz-floss/frequenz-api-microgrid/assets/1031485/1dcd5512-f036-4890-9c2b-1c448b05cfee)

GitHub should really do something about those review comments, they are very easy to miss and I just realized they don't even have a way to put a link to them!

tiyash-basu-frequenz commented:
>I guess you didn't saw the review comment above?

You are right, I missed it.

It was a good suggestion though. I fixed the line you had linked. Plus, I also found another line in the release notes that referred to v0.3, and removed it.
